api/services/cts.go

```diff
@@ -44,6 +45,42 @@ func (c *ServiceClient) Close() {
 	c.taskConn.Close()
 }
 
+/////////////////////////////
+//      Health Check       //
+/////////////////////////////
+
+func (c *ServiceClient) HealthCheck(ctx context.Context) (bool, error) {
+	healthClient := grpc_health_v1.NewHealthClient(c.taskConn)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	// Health check
+	resp, err := healthClient.Check(ctx, &grpc_health_v1.HealthCheckRequest{
+		Service: "task.TaskService",
+	})
+
+	if err != nil {
+		return false, err
+	}
+
+	if resp.Status == grpc_health_v1.HealthCheckResponse_SERVING {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+func (c *ServiceClient) DetailedHealthCheck(ctx context.Context, args *task.DetailedHealthCheckRequest) (*task.DetailedHealthCheckResponse, error) {
+	ctx, cancel := context.WithTimeout(ctx, DEFAULT_PROCESS_DEADLINE)
+	defer cancel()
+	resp, err := c.taskService.DetailedHealthCheck(ctx, args)
+	if err != nil {
```

Comment:
Should use `err` only if the health check itself failed due to an internal error. Otherwise, should check if the `UnhealthyReasons` list is empty and respond based on that.